src/app/public/modules/toggle/toggle.component.spec.ts

```diff
@@ -0,0 +1,740 @@
+}
+
+/** Simple test component with multiple toggles. */
+@Component(({
+  template: `
+    <sky-toggle>
+      <sky-toggle-label>
+        Option 1
+      </sky-toggle-label>
+    </sky-toggle>
+    <sky-toggle>Option 2</sky-toggle>
+  `
+}))
+class MultipleTogglesComponent { }
+
+/** Simple test component with tabIndex */
+@Component({
+  template: `
+    <sky-toggle [tabindex]="customTabIndex" [disabled]="isDisabled">
+    </sky-toggle>`
+})
+class ToggleWithTabIndexComponent {
+  public customTabIndex: number = 7;
+  public isDisabled: boolean = false;
+}
+
+/** Simple test component with an aria-label set. */
+@Component({
+  template: `<sky-toggle label="Super effective"></sky-toggle>`
+})
+class ToggleWithAriaLabelComponent { }
+
+/** Simple test component with an aria-label set. */
+@Component({
+  template: `<sky-toggle labelledBy="some-id"></sky-toggle>`
+})
+class ToggleWithAriaLabelledbyComponent { }
+
+/** Simple test component with name attribute */
+@Component({
+  template: `<sky-toggle name="test-name"></sky-toggle>`
+})
+class ToggleWithNameAttributeComponent { }
+
+/** Simple test component with change event */
+@Component({
+  template: `<sky-toggle (change)="lastEvent = $event"></sky-toggle>`
+})
+class ToggleWithChangeEventComponent {
+  public lastEvent: SkyToggleChange;
+}
+
+/** Simple test component with OnPush change detection */
+@Component({
+  template: `
+  <div>
+    <sky-toggle
+      id="simple-toggle"
+      [(ngModel)]="isToggled">
+    </sky-toggle>
+  </div>`,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+class ToggleWithOnPushChangeDetectionComponent {
+  public isToggled: boolean = false;
+  constructor(public ref: ChangeDetectorRef) { }
+}
+
+describe('Toggle component', () => {
+  let fixture: ComponentFixture<any>;
+
+  function createEvent(eventName: string) {
+    let evt = document.createEvent('CustomEvent');
+    evt.initEvent(eventName, false, false);
+    return evt;
+  }
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        ToggleWithAriaLabelComponent,
+        ToggleWithAriaLabelledbyComponent,
+        ToggleWithChangeEventComponent,
+        ToggleWithFormDirectivesComponent,
+        ToggleWithNameAttributeComponent,
+        ToggleWithOnPushChangeDetectionComponent,
+        ToggleWithTabIndexComponent,
+        ToggleWithReactiveFormComponent,
+        MultipleTogglesComponent,
+        SingleToggleComponent
+      ],
+      imports: [
+        BrowserModule,
+        FormsModule,
+        ReactiveFormsModule,
+        SkyToggleModule
+      ]
+    });
+  });
+
+  describe('basic behaviors', () => {
+    let toggleDebugElement: DebugElement;
+    let toggleNativeElement: HTMLElement;
+    let toggleInstance: SkyToggleComponent;
+    let testComponent: SingleToggleComponent;
+    let inputElement: HTMLInputElement;
+    let labelElement: HTMLLabelElement;
+
+    beforeEach(async(() => {
```

Comment:
Can we do this with `fakeAsync` and `tick`?

Reply:
`fakeAsync` is better when we can. So I would have that same question throughout your tests.